File: week1/backend/app/api/v1/aggregations.py
"""聚合查询路由：``/api/v1/tags`` 与 ``/api/v1/assignees``。

对应 spec §5.3 的辅助接口。
"""

from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_db_session
from app.schemas.common import ApiResponse
from app.services import aggregation_service
from app.utils.responses import success

router = APIRouter(tags=["aggregations"])


@router.get(
    "/tags",
    response_model=ApiResponse[list[str]],
    summary="获取所有已使用的标签",
)
def list_tags(db: Session = Depends(get_db_session)) -> ApiResponse[list[str]]:
    return success(aggregation_service.list_tags(db))


@router.get(
    "/assignees",
    response_model=ApiResponse[list[str]],
    summary="获取所有已使用的负责人",
)
def list_assignees(db: Session = Depends(get_db_session)) -> ApiResponse[list[str]]:
    return success(aggregation_service.list_assignees(db))

File: week1/frontend/src/api/aggregations.ts
/**
 * 聚合接口：标签 / 负责人（spec §5.3）。
 */
import type { AxiosRequestConfig } from 'axios'

import { request } from '@/api/request'

export function listTags(config?: AxiosRequestConfig): Promise<string[]> {
  return request.get<unknown, string[]>('/tags', config)
}

export function listAssignees(config?: AxiosRequestConfig): Promise<string[]> {
  return request.get<unknown, string[]>('/assignees', config)
}
